docs/docs/routing.md

```diff
@@ -0,0 +1,243 @@
+
+While it is certainly possible to create your own object to manage access to the History API, in this guide we'll use the popular [history](https://github.com/mjackson/history) package.
+
+```bash
+npm install history --save
+```
+
+As the History API is global, our object will also need to be accessible everywhere.
+
+```js
+import createBrowserHistory from 'history/createBrowserHistory';
+
+const history = createBrowserHistory();
+```
+
+This guide will use the following methods and properties from the `history` object:
+
+```js
+// Get the current location.
+const location = history.location
+
+// Listen for changes to the current location.
+const unlisten = history.listen((location) => {
+  // location is an object like window.location
+  console.log(location.pathname)
+})
+
+// Change current location, adding a new entry to the
+// browser history.
+history.push('/home')
+```
+
+You can view the documentation for the full `history` object API at the [history](https://github.com/mjackson/history) package's page.
+
+## Links
+
+By default, HTML `<a>` elements will always cause the page to reload when clicked. If you'd like to update the URL using the History API, you'll need to cancel the default behaviour and then implement an alternative.
+
+You could do this by adding an `onClick` handler to each individual `<a>` element, but since most links within an app will share the same logic, people often create a component for them. Instances of this component behave exactly like `<a>` elements, except that instead of refreshing the page, they will update the URL in-place using `history.push()`:
+
+```js
+class Link extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  handleClick(event) {
+    // Instead of letting the `<a>` element handle
+    // onClick, do it here
+    if (this.props.onClick) {
+      this.props.onClick(event);
+    }
+
+    // Don't navigate if this.props.onClick called
+    // e.preventDefault()
+    if (event.defaultPrevented) {
+      return;
+    }
+
+    // If target prop is set (e.g. to "_blank"),
+    // let browser handle link.
+    if (this.props.target) {
+      return;
+    }
+    
+    const isNotLeftClick = event.button !== 0;
+    const isModified = !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
+    if (isModified || isNotLeftClick) {
+      return;
+    }
+
+    // Ensure the browser doesn't navigate
+    event.preventDefault();
+    
+    // history.push() expects an object that takes the
+    // same shape as `window.location`
+    const pathname = this.props.href.split('?')[0];
+    const search = this.props.href.slice(pathname.length);
+    const location = { pathname, search };
+    
+    // Update the browser URL and notify any listeners
+    // added via history.listen()
+    history.push(location);
+  }
+
+  render() {
+    return (
+      <a {...this.props} onClick={this.handleClick}>
```

Comment:
Minor nit: technically `{this.props.children}` is not required if you already spread props.
But I guess that's clearer.